OpenRA.Mods.Common/Effects/Bullet.cs

```diff
@@ -171,12 +171,12 @@ public void Tick(World world)
 
 		public IEnumerable<IRenderable> Render(WorldRenderer wr)
 		{
-			if (info.ContrailLength > 0)
-				yield return contrail;
-
```

Comment:
I haven't tested this, but I think the idea here was that bullet projectiles should be able to display a contrail even if the bullet itself has no image (I think RA snipers used this).